app/client/src/widgets/TableWidget/widget/index.tsx

```diff
@@ -146,7 +146,9 @@ class TableWidget extends BaseWidget<TableWidgetProps, WidgetState> {
         Cell: (props: any) => {
           const rowIndex: number = props.cell.row.index;
           const data = this.props.filteredTableData[rowIndex];
-          const originalIndex = data?.__originalIndex__ || rowIndex;
+          const originalIndex = !isNil(data?.__originalIndex__)
+            ? data.__originalIndex__
+            : rowIndex;
```

Comment:
This should work as well

```suggestion
          const originalIndex = data?.__originalIndex__ ?? rowIndex;
```

Reply:
I made the above suggested changes.